### PR TITLE
[android] build ot-ctl on Android.mk

### PR DESCRIPTION
**Pull request by bukepo**
* Add build ot-ctl
* Enable platform netif and platform UDP.

### CODE REVIEW

**Comment by bukepo** [comment]
@gjc any comments?

**Comment by gjc** [comment]
Hi - I think this one is for @gjc13. Thank you

File: Android.mk [hunk 1]
```diff
@@ -59,8 +59,14 @@ OPENTHREAD_COMMON_FLAGS                                          += \
     $(NULL)
 endif
 
-ifneq ($(USE_OTBR_DAEMON), 1)
-OPENTHREAD_COMMON_FLAGS  += -DOPENTHREAD_CONFIG_UDP_FORWARD_ENABLE=1
+ifeq ($(USE_OTBR_DAEMON), 1)
+OPENTHREAD_COMMON_FLAGS                                          += \
```
Comment (gjc13) [inline Android.mk]: It seems we're both adding these flags
https://github.com/openthread/openthread/pull/4568.
So maybe I can close my PR?

Comment (bukepo) [reply Android.mk]: No, your PR adds default value, there will be no duplicate definitions.

**Pull request merged**